### PR TITLE
[GPU] Add binary_sub post-op for oneDNN

### PR DESCRIPTION
**Pull request by jade-cho**
### Tickets:
 - *94581*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 1]
```diff
@@ -1076,6 +1078,10 @@ void program_node::init_onednn_primitive_attributes() {
                     post_ops.append_binary(dnnl::algorithm::binary_add, in_desc);
                     update_onednn_post_op_list(onednn_post_op_type::binary_add, dep_idx);
                 }
+            } else if (desc.typed_desc<eltwise>()->mode == eltwise_mode::sub) {
+                dnnl::memory::desc in_desc = onednn::layout_to_memory_desc(in);
+                post_ops.append_binary(dnnl::algorithm::binary_sub, in_desc);
+                update_onednn_post_op_list(onednn_post_op_type::binary_sub, dep_idx);
             } else {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:1085]: I think we are missing some assertion logic here. What if mode is not `sum`, nor `sub`?

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 2]
```diff
@@ -1073,6 +1079,11 @@ void program_node::init_onednn_primitive_attributes() {
                     attrs->set_output_scales(mask, {DNNL_RUNTIME_F32_VAL});
                     update_onednn_post_op_list(onednn_post_op_type::scale, dep_idx);
                 }
+            } else {
+                std::stringstream error_msg;
+                error_msg << "Unsupported eltwise mode: " << static_cast<int>(desc.typed_desc<eltwise>()->mode) << ". ";
+                error_msg << desc.desc->id << " is fused node of " + this->id() + ".";
+                OPENVINO_ASSERT(false, error_msg.str());
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:1086]: could you check that there is no diff from dryrun or partial run with this change?

Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:1086]: No difference at dryrun test.

File: src/plugins/intel_gpu/src/graph/program_node.cpp (hunk 1 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/program_node.cpp]: Added

**Review by byungilm — APPROVED** [review]

**Pull request merged**